util/retrieve-cluster-logs.py

```diff
@@ -0,0 +1,213 @@
+import gzip
+import json
+import logging
+import os
+import sys
+import tarfile
+import tempfile
+from datetime import datetime
+
+import argparse
+import boto3
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__file__)
+
+
+def err_and_exit(message):
+    """Log the given error message and exit nonzero."""
+    LOGGER.error(message)
+    sys.exit(1)
+
+
+def get_default_bucket_prefix(cluster_name):
+    """Get default keypath to store logs under in an s3 bucket for the given cluster."""
+    return "{cluster_name}-logs-{end_time}".format(
+        cluster_name=cluster_name, end_time=datetime.now().strftime("%Y%m%d%H%M%S")
+    )
+
+
+def get_log_group(args):
+    """Get log group for args.cluster."""
+    log_group_name = "/aws/parallelcluster/{}".format(args.cluster)
+    logs = boto3.client("logs", region_name=args.region)
```

Comment:
if you don't want to pass this region around you could set it as an env variable named AWS_DEFAULT_REGION. boto3 will pick that up automatically.

Reply:
Honestly I'd prefer to specify it as a kwarg. Do we usually set the env? If so I'll change it.

Reply:
In the CLI we are transitioning to this approach since passing region, proxy and credentials args to all functions calling boto3 was probably not the best approach. At the same time you might forget to set this parameters when initializing a boto3 client. 
Of course at the same time you might argue that this hides the fact that this boto3 calls are using specific parameters loaded from the environment.

Feel free to leave it as is here if you find this approach reasonable for this use case.